birdhouse/components/cowbird/docker-compose-extra.yml

```diff
@@ -36,7 +36,7 @@ services:
       - ./components/cowbird/config/cowbird/cowbird.ini:/opt/local/src/cowbird/config/cowbird.ini
       # even if not running tasks here, they must be registered to send them off to the right place!
       - ./components/cowbird/config/cowbird/celeryconfig.py:/opt/local/src/cowbird/config/celeryconfig.py
-      - "${DATA_PERSIST_ROOT}/${USER_WORKSPACES}:/${USER_WORKSPACES}"
+      - "${USER_WORKSPACES}:/user_workspaces"
```

Comment:
As mentioned by @ChaamC, this could easily break Cowbird.
Something tricky about these volumes is that Cowbird creates dynamic hardlinks between different locations that must resolve properly once mounted. However, even though Cowbird sees the mounted location `/user_workspaces` inside its docker container, the hardlinks it must create must resolve properly on the host side, such that once this link is mounted elsewhere such as in Jupyter user-specific dockers, the file still resolves correctly with a reduced/nested directory of the specific user-workspace. 

@ChaamC and I have had a lot of discussions and did many exploration tests to try to resolve this already complicated problem. I would prefer not to introduce customizable paths here, to avoid Cowbird having to deal with complicated inverse-volume-mapping resolution to create the links correctly.

Something that could be considered is to have some kind of `DATA_PERSIST_SHARED_DIR` (instead of  `DATA_PERSIST_ROOT`) where stuff like user-workspaces and wps-outputs that must share locations between services are all nested under. However, the `${DATA_PERSIST_SHARED_DIR}/user_workspaces` and `${DATA_PERSIST_SHARED_DIR}/wps_outputs` would be hardcoded to ensure they work properly.